SDL3/CategoryStorage.md

```diff
@@ -1,6 +1,155 @@
 # CategoryStorage
 
-SDL storage container management.
+The storage API is a high-level API designed to abstract away the portability issues that come up when using something lower-level (in SDL's case, this sits on top of SDL_filesystem). It is significantly more restrictive than a typical filesystem API, for a number of reasons:
+
+1. **What to Access:** A common pitfall with existing filesystem APIs is the assumption that all storage is monolithic. However, many other platforms (game consoles in particular) are more strict about what _type_ of filesystem is being accessed; for example, game content and user data are usually two separate storage devices with entirely different characteristics (and possibly different low-level APIs altogether!).
+2. **How to Access:** Another common mistake is applications assuming that all storage is universally writeable - again, many platforms treat game content and user data as two separate storage devices, and only user data is writeable while game content is read-only.
+3. **When to Access:** The most common portability issue with filesystem access is _timing_ - you cannot always assume that the storage device is always accessible all of the time, nor can you assume that there are no limits to how long you have access to a particular device.
+
+Consider the following example:
+
+```
+void ReadGameData(void)
+{
+    extern char** fileNames;
+    extern size_t numFiles;
+    for (size_t i = 0; i < numFiles; i += 1) {
+        FILE *data = fopen(fileNames[i], "rwb");
+        if (data == NULL) {
+            // Something bad happened!
+        } else {
+            // A bunch of stuff happens here
+            fclose(data);
+        }
+    }
+}
+
+void ReadSave(void)
+{
+    FILE *save = fopen("saves/save0.sav", "rb");
+    if (save == NULL) {
+        // Something bad happened!
+    } else {
+        // A bunch of stuff happens here
+        fclose(save);
+    }
+}
+
+void WriteSave(void)
+{
+    FILE *save = fopen("saves/save0.sav", "wb");
+    if (save == NULL) {
+        // Something bad happened!
+    } else {
+        // A bunch of stuff happens here
+        fclose(save);
+    }
+}
+```
+
+Going over the bullet points again:
+
+1. **What to Access:** This code accesses a global filesystem; game data and saves are all presumed to be in the game's installation folder.
```

Comment:
Actually, they're presumed to be relative to the current directory, which is a terrible assumption.

Reply:
Even better!!

Reply:
```suggestion
1. **What to Access:** This code accesses a global filesystem; game data and saves are all presumed to be in the current working directory (which may or may not be the game's installation folder!).
```